pkg/pillar/types/global.go

```diff
@@ -114,6 +115,132 @@ func (gs *GlobalStatus) UpdateItemValuesFromGlobalConfig(gc GlobalConfig) {
 // Agents subscribe to this info to get at least the log levels
 // A value of zero means we should use the default
 // All times are in seconds.
+type ConfigItemType uint8
+
+const (
+	CONFIGITEM_TYPE_INT ConfigItemType = iota
+	CONFIGITEM_TYPE_BOOL
+	CONFIGITEM_TYPE_STRING
+	CONFIGITEM_TYPE_TRISTATE
+)
+
+type ConfigItemSpec struct {
+	Key      string
+	ItemType ConfigItemType
+
+	intMin     uint32
+	intMax     uint32
+	intDefault uint32
+
+	stringDefault   string
+	boolDefault     bool
+	triStateDefault TriState
+}
+
+type ConfigItemSpecMap struct {
+	itemMap map[string]ConfigItemSpec
+}
+
+func (configMap *ConfigItemSpecMap) AddIntItem(key string, defaultInt uint32, min uint32, max uint32) {
+	log.Debugf("Adding int item %s", key)
+	if defaultInt < min || defaultInt > max {
+		log.Fatalf("Adding int item %s failed. Value does not meet given min/max criteria", key)
+	}
+	configItem := ConfigItemSpec{
+		ItemType:   CONFIGITEM_TYPE_INT,
+		Key:        key,
+		intDefault: defaultInt,
+		intMin:     min,
+		intMax:     max,
+	}
+	configMap.itemMap[key] = configItem
+	log.Debugf("Added int item %s", key)
+}
+
+func (configMap *ConfigItemSpecMap) AddBoolItem(key string, defaultBool bool) {
+	log.Debugf("Adding bool item %s", key)
+	configItem := ConfigItemSpec{
+		ItemType:    CONFIGITEM_TYPE_BOOL,
+		Key:         key,
+		boolDefault: defaultBool,
+	}
+	configMap.itemMap[key] = configItem
+	log.Debugf("Added bool item %s", key)
+}
+
+func (configMap *ConfigItemSpecMap) AddStringItem(key string, defaultString string) {
+	log.Debugf("Adding string item %s", key)
+	configItem := ConfigItemSpec{
+		ItemType:      CONFIGITEM_TYPE_STRING,
+		Key:           key,
+		stringDefault: defaultString,
+	}
+	configMap.itemMap[key] = configItem
+	log.Debugf("Added string item %s", key)
+}
+
+func (configMap *ConfigItemSpecMap) AddTriStateItem(key string, defaultTriState TriState) {
+	log.Debugf("Adding tristate item %s", key)
+	configItem := ConfigItemSpec{
+		Key:             key,
+		ItemType:        CONFIGITEM_TYPE_TRISTATE,
+		triStateDefault: defaultTriState,
+	}
+	configMap.itemMap[key] = configItem
+	log.Debugf("Added tristate item %s", key)
+}
+type GlobalConfigItemSpec struct {
```

Comment:
Rename as GlobalConfigItem.
This one is for actual config items.. Not their specification deinitions

ConfigItemSpec / ConfigItemSpecMap ---> These hold the specifications of all supported items

GlobalConfigItem / GlobalConfigItemMap --> These hold the Actual configurations. They use the spec to parse the item and store it in GlobalConfigItem,